linode/nb/framework_resource.go

```diff
@@ -0,0 +1,269 @@
+		Label:              data.Label.ValueStringPointer(),
+		ClientConnThrottle: &clientConnThrottle,
+	}
+
+	if !data.Tags.IsNull() {
+		resp.Diagnostics.Append(data.Tags.ElementsAs(ctx, &createOpts.Tags, false)...)
+		if resp.Diagnostics.HasError() {
+			return
+		}
+	}
+
+	nodebalancer, err := client.CreateNodeBalancer(ctx, createOpts)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error creating a Linode NodeBalancer",
+			err.Error(),
+		)
+		return
+	}
+
+	resp.Diagnostics.Append(data.parseComputedAttrs(ctx, nodebalancer)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func (r *Resource) Read(
+	ctx context.Context,
+	req resource.ReadRequest,
+	resp *resource.ReadResponse,
+) {
+	var data NodebalancerModel
+	client := r.Meta.Client
+
+	resp.Diagnostics.Append(req.State.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	nodebalancer, err := client.GetNodeBalancer(ctx, int(data.ID.ValueInt64()))
+	if err != nil {
+		if lerr, ok := err.(*linodego.Error); ok && lerr.Code == 404 {
+			resp.Diagnostics.AddWarning(
+				"Nodebalancer",
+				fmt.Sprintf("removing Linode NodeBalancer ID %v from state because it no longer exists", data.ID.ValueInt64()),
+			)
+			resp.State.RemoveResource(ctx)
+			return
+		}
+		resp.Diagnostics.AddError(
+			fmt.Sprintf("Failed to get nodebalancer %v", data.ID.ValueInt64()),
+			err.Error(),
+		)
+	}
+
+	resp.Diagnostics.Append(data.parseComputedAttrs(ctx, nodebalancer)...)
+	resp.Diagnostics.Append(data.parseNonComputedAttrs(ctx, nodebalancer)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func (r *Resource) Update(
+	ctx context.Context,
+	req resource.UpdateRequest,
+	resp *resource.UpdateResponse,
+) {
+	var plan, state NodebalancerModel
+	client := r.Meta.Client
+
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	isEqual := state.Label.Equal(plan.Label) &&
+		state.ClientConnThrottle.Equal(plan.ClientConnThrottle) &&
+		state.Tags.Equal(plan.Tags)
+
+	if !isEqual {
+		clientConnThrottle := int(plan.ClientConnThrottle.ValueInt64())
+		updateOpts := linodego.NodeBalancerUpdateOptions{
+			Label:              plan.Label.ValueStringPointer(),
+			ClientConnThrottle: &clientConnThrottle,
+		}
+		resp.Diagnostics.Append(plan.Tags.ElementsAs(ctx, &updateOpts.Tags, false)...)
+		if resp.Diagnostics.HasError() {
+			return
+		}
+		nodebalancer, err := client.UpdateNodeBalancer(ctx, int(plan.ID.ValueInt64()), updateOpts)
+		if err != nil {
+			resp.Diagnostics.AddError(
+				fmt.Sprintf("Failed to update Nodebalancer %v", plan.ID.ValueInt64()),
+				err.Error(),
+			)
+			return
+		}
+
+		resp.Diagnostics.Append(plan.parseComputedAttrs(ctx, nodebalancer)...)
+		resp.Diagnostics.Append(resp.State.Set(ctx, &plan)...)
+	}
```

Comment:
> An error is returned unless every null or known value in the request plan is saved exactly as-is into the response state. Only unknown plan values can be modified.

https://developer.hashicorp.com/terraform/plugin/framework/resources/update#caveats
I think we should technically save `plan` outside of the if block in this case.

```suggestion
		resp.Diagnostics.Append(plan.parseComputedAttrs(ctx, nodebalancer)...)
		if resp.Diagnostics.HasError() {
			return
		}
	}
	resp.Diagnostics.Append(resp.State.Set(ctx, &plan)...)
```

Reply:
I think this is not necessary though.